### ISSUE TITLE
Create seperate cargo workspaces

### ISSUE THREAD

**Issue by Nutomic** [LemmyNet/lemmy#809]
https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html

We should create seperate workspaces to speed up compilation and organize the code better. I managed to reduce compilation time from ~80s to ~60s with the following dependency tree:
```
lemmy_server
├── lemmy_utils (containing settings and regex functions)
├── lemmy_db (just the database, with as few dependencies as possible)
```

The important part here is that db doesnt depend on utils, so that they can be compiled in parallel. In general it will help if we can identify more parts of our code that dont depend on the database, and put them in seperate crates for parallel compilation.

Right now it would be too complicated to implement this, I will wait until federation is merged into master.

Edit: It should also help to move serde objects to a seperate crate.

**Comment by dessalines** [comment]
U still want this one open @Nutomic ? I know you said you'd maybe wanna extract some more.

**Comment by Nutomic** [comment]
Yes at some point I can move api and apub into separate workspaces. But it will probably not do much for compilation time, so it seems rather low priority.
